pkg/test/framework/components/echo/factory.go

```diff
@@ -0,0 +1,41 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package echo
+
+import (
+	"fmt"
+
+	"istio.io/istio/pkg/test/framework/components/cluster"
+	"istio.io/istio/pkg/test/framework/resource"
+)
+
+// FactoryFunc can be used by a builder to produce instances from configs
+type FactoryFunc func(ctx resource.Context, config []Config) (Instances, error)
+
+var factoryRegistry = map[cluster.Kind]FactoryFunc{}
```

Comment:
Is there any difference between this and using `make(map...)`? For some reason I thought `make` was preferred.

Reply:
Only if you need to specify capacity for some perf reason.